src/environmentd/src/http/sql.rs

```diff
@@ -571,14 +575,11 @@ trait ResultSender: Send {
     /// Reports whether the client supports streaming SUBSCRIBE results.
     fn allow_subscribe(&self) -> bool;
 
-    async fn await_rows<F, R>(&mut self, f: F) -> Result<R, Error>
-    where
-        F: Future<Output = R> + Send,
-    {
-        tokio::select! {
-            e = self.connection_error() => Err(e),
-            r = f => Ok(r),
-        }
+    /// Emits a streaming notice if the sender supports it.
+    ///
+    /// Does nothing if `SUPPORTS_STREAMING_NOTICES` is false.
+    async fn emit_streaming_notice(&mut self, _: AdapterNotice) -> Result<(), Error> {
+        Ok(())
```

Comment:
If this is incorrectly called the notice will be dropped. Perhaps this function should `unreachable!()` instead?

Reply:
Sure, done!